src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/EngineCloseTest.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2014 Feedzai
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.feedzai.commons.sql.abstraction.engine.impl.abs;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import com.feedzai.commons.sql.abstraction.ddl.DbEntity;
+import com.feedzai.commons.sql.abstraction.engine.*;
+import com.feedzai.commons.sql.abstraction.engine.testconfig.DatabaseConfiguration;
+import com.feedzai.commons.sql.abstraction.engine.testconfig.DatabaseTestUtil;
+import mockit.*;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.slf4j.LoggerFactory;
+
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Properties;
+
+import static com.feedzai.commons.sql.abstraction.engine.EngineTestUtils.buildEntity;
+import static com.feedzai.commons.sql.abstraction.engine.configuration.PdbProperties.*;
+
+
+/**
+ * Tests closing a {@link DatabaseEngine} to make sure all resources are cleaned up correctly.
+ *
+ * It tests with each schema policy.
+ *
+ * @author David Fialho (david.fialho@feedzai.com)
+ * @since 2.1.13
+ */
+@RunWith(Parameterized.class)
+public class EngineCloseTest {
+
+    protected DatabaseEngine engine;
+    protected Properties properties;
+
+    // Parameters
+    protected DatabaseConfiguration config;
+    protected String schemaPolicy;
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() throws Exception {
+
+        final Collection<DatabaseConfiguration> configurations = DatabaseTestUtil.loadConfigurations();
+        final Collection<String> schemaPolicies = Arrays.asList(
+                "drop-create",
```

Comment:
This is OK for now and especially for this pull-request, but in the future, we should create an Enum for this instead of using strings everywhere (and then using it across the project).

Reply:
Totally agree :+1: